test/run-spec-wasm2c.py

```diff
@@ -466,7 +466,7 @@ def Compile(cc, c_filename, out_dir, *cflags):
                  '-Wno-infinite-recursion',
                  '-fno-optimize-sibling-calls',
                  '-frounding-math', '-fsignaling-nans',
-                 '-std=c99', '-D_DEFAULT_SOURCE']
+                 '-std=c11', '-D_DEFAULT_SOURCE']
```

Comment:
On thinking further, rather than a complete switch to c11, maybe we could keep the code working for c99 in general, but use c11 in tests that need threads/atomics.

 @keithw I think for this, my plan forward is to try to update the test infrastructure to use c11 only when specified as a flag in the test. Does this make sense as path forward?

Reply:
Does this + the new docs address this sufficiently?